File: lib/models/breakpoint_response.dart
import 'captured_exchange.dart';

/// User decision for a suspended request (UI → core), RF3.2.
enum BreakpointAction { proceed, cancel }

class BreakpointResponse {
  final String breakpointId;
  final BreakpointAction action;
  final String? modifiedMethod;
  final String? modifiedUrl;
  final List<HttpHeader>? modifiedHeaders;
  final String? modifiedBody;
  final int? modifiedStatus;
  final DateTime timestamp;

  const BreakpointResponse({
    required this.breakpointId,
    required this.action,
    this.modifiedMethod,
    this.modifiedUrl,
    this.modifiedHeaders,
    this.modifiedBody,
    this.modifiedStatus,
    required this.timestamp,
  });

  Map<String, dynamic> toMap() => {
    'breakpointId': breakpointId,
    'action': action.name,
    'modifiedMethod': modifiedMethod,
    'modifiedUrl': modifiedUrl,
    'modifiedHeaders': modifiedHeaders
        ?.map((h) => {'name': h.name, 'value': h.value})
        .toList(),
    'modifiedBody': modifiedBody,
    'modifiedStatus': modifiedStatus,
    'timestamp': timestamp.toIso8601String(),
  };
}

File: Sources/RoxProxy/Models/BreakpointResponse.swift
import Foundation

/// Represents the user's response to a breakpoint from Flutter to backend via WebSocket
struct BreakpointResponse: Codable {
    let breakpointId: UUID
    let action: BreakpointAction
    let modifiedMethod: String?
    let modifiedUrl: String?
    let modifiedHeaders: [String: String]?
    let modifiedBody: String?
    let timestamp: Date
    
    enum BreakpointAction: String, Codable {
        case proceed
        case cancel
    }
    
    init(
        breakpointId: UUID,
        action: BreakpointAction,
        modifiedMethod: String? = nil,
        modifiedUrl: String? = nil,
        modifiedHeaders: [String: String]? = nil,
        modifiedBody: String? = nil,
        timestamp: Date = Date()
    ) {
        self.breakpointId = breakpointId
        self.action = action
        self.modifiedMethod = modifiedMethod
        self.modifiedUrl = modifiedUrl
        self.modifiedHeaders = modifiedHeaders
        self.modifiedBody = modifiedBody
        self.timestamp = timestamp
    }
}